src/Symfony/Component/Ldap/Adapter/ExtLdap/Query.php

```diff
@@ -203,4 +194,67 @@ private function resetPagination()
             ldap_set_option($con, \LDAP_OPT_SERVER_CONTROLS, $ctl);
         }
     }
+
+    protected function controlPagedResult($con, $pageSize, $isCritical = false, $cookie = '')
+    {
+        if (\PHP_VERSION_ID < 70300) {
+            return ldap_control_paged_result($con, $pageSize, $isCritical, $cookie);
+        }
+        $this->options['serverctrls'] = [
```

Comment:
IMO, we should avoid changing the state of `$this->options`. This might have unintended side effects as it changes the state of the class. This `serverctrls` should rather be handled using local state (return it from this method and store it in a local variable in the main query, or maybe inline that logic directly with the code calling the function by passing it the cookie